service/src/test/java/uk/gov/hmcts/reform/fpl/service/GeneratedOrderServiceTest.java

```diff
@@ -20,203 +24,206 @@
 import uk.gov.hmcts.reform.fpl.model.common.Element;
 import uk.gov.hmcts.reform.fpl.model.common.JudgeAndLegalAdvisor;
+import uk.gov.hmcts.reform.fpl.service.config.LookupTestConfig;
 import uk.gov.hmcts.reform.fpl.service.time.Time;
+import uk.gov.hmcts.reform.fpl.utils.FixedTimeConfiguration;
 
-import java.io.IOException;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.format.FormatStyle;
+import java.util.HashMap;
 import java.util.Map;
+import java.util.stream.Stream;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.fpl.enums.GeneratedOrderSubtype.FINAL;
+import static uk.gov.hmcts.reform.fpl.enums.GeneratedOrderSubtype.INTERIM;
 import static uk.gov.hmcts.reform.fpl.enums.GeneratedOrderType.BLANK_ORDER;
 import static uk.gov.hmcts.reform.fpl.enums.GeneratedOrderType.CARE_ORDER;
 import static uk.gov.hmcts.reform.fpl.enums.JudgeOrMagistrateTitle.HER_HONOUR_JUDGE;
 import static uk.gov.hmcts.reform.fpl.utils.CaseDataGeneratorHelper.createOrders;
 import static uk.gov.hmcts.reform.fpl.utils.DocumentManagementStoreLoader.document;
 
 @ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {
+    FixedTimeConfiguration.class, LookupTestConfig.class, DateFormatterService.class, GeneratedOrderService.class
+})
 class GeneratedOrderServiceTest {
-    private static final String LOCAL_AUTHORITY_CODE = "example";
-    private static final String LOCAL_AUTHORITY_NAME = "Example Local Authority";
-    private static final String COURT_NAME = "Example Court";
-    private static final String COURT_EMAIL = "example@court.com";
-    private static final String COURT_CONFIG = String.format("%s=>%s:%s", LOCAL_AUTHORITY_CODE, COURT_NAME,
-        COURT_EMAIL);
-    private static final String LA_NAME_CONFIG = String.format("%s=>%s", LOCAL_AUTHORITY_CODE, LOCAL_AUTHORITY_NAME);
-    private static final LocalDateTime NOW = LocalDateTime.now();
-
-    private final Time time = () -> NOW;
-
-    private DateFormatterService dateFormatterService = new DateFormatterService();
-    private HmctsCourtLookupConfiguration hmctsCourtLookupConfiguration = new HmctsCourtLookupConfiguration(
-        COURT_CONFIG);
-    private LocalAuthorityNameLookupConfiguration localAuthorityNameLookupConfiguration =
-        new LocalAuthorityNameLookupConfiguration(LA_NAME_CONFIG);
 
+    @Autowired
+    private Time time;
+
+    @Autowired
+    private DateFormatterService dateFormatterService;
+
+    @Autowired
     private GeneratedOrderService service;
 
-    @BeforeEach
-    void setup() {
-        this.service = new GeneratedOrderService(dateFormatterService, hmctsCourtLookupConfiguration,
-            localAuthorityNameLookupConfiguration, time);
-    }
+    @Nested
+    class C21Tests {
 
-    @Test
-    void shouldAddDocumentToOrderTypeAndDocumentObjectWhenDocumentExists() throws IOException {
-        Document document = document();
+        C21Tests() {
+            //NO - OP
```

Comment:
Do we need that comment?

Reply:
Can probably delete the whole constructor tbh